src/Features/Core/Portable/PullMemberUp/MembersPuller.cs

```diff
@@ -0,0 +1,408 @@
+            // will be add to the destination interface.
+            return getterOrSetter?.DeclaredAccessibility == Accessibility.Public ? getterOrSetter : null;
+        }
+
+        private IMethodSymbol CreatePublicGetterAndSetter(IMethodSymbol getterOrSetter)
+        {
+            // Create a public getter/setter since user is trying to pull a non-public property to an interface.
+            // If getterOrSetter is null, it means this property doesn't have a getter/setter, so just don't generate it.
+            return getterOrSetter == null
+                ? getterOrSetter
+                : CodeGenerationSymbolFactory.CreateMethodSymbol(getterOrSetter, accessibility: Accessibility.Public);
+        }
+
+        private async Task<Solution> PullMembersIntoInterfaceAsync(
+            Document document,
+            PullMembersUpAnalysisResult result,
+            CancellationToken cancellationToken)
+        {
+            var solution = document.Project.Solution;
+            var solutionEditor = new SolutionEditor(solution);
+            var codeGenerationService = document.Project.LanguageServices.GetRequiredService<ICodeGenerationService>();
+            var destinationSyntaxNode = await codeGenerationService.FindMostRelevantNameSpaceOrTypeDeclarationAsync(
+                document.Project.Solution, result.Destination, default, cancellationToken).ConfigureAwait(false);
+            var (editorMap, syntaxMap) = await InitializeEditorMapsAndSyntaxMapAsync(result, solution, solutionEditor, destinationSyntaxNode, cancellationToken).ConfigureAwait(false);
+            var symbolsToPullUp = result.MemberAnalysisResults.
+                SelectAsArray(analysisResult =>
+                {
+                    if (analysisResult.Member is IPropertySymbol propertySymbol)
+                    {
+                        if (analysisResult.ChangeOriginalToPublic)
+                        {
+                            // We are pulling a non-public property, change its getter/setter to public and itself to be public.
+                            return CodeGenerationSymbolFactory.CreatePropertySymbol(
+                                propertySymbol,
+                                accessibility: Accessibility.Public,
+                                getMethod: CreatePublicGetterAndSetter(propertySymbol.GetMethod),
+                                setMethod: CreatePublicGetterAndSetter(propertySymbol.SetMethod));
+                        }
+                        else
+                        {
+                            // We are pulling a public property, filter the non-public getter/setter.
+                            return CodeGenerationSymbolFactory.CreatePropertySymbol(
+                                propertySymbol,
+                                getMethod: FilterGetterOrSetter(propertySymbol.GetMethod),
+                                setMethod: FilterGetterOrSetter(propertySymbol.SetMethod));
```

Comment:
nit: consider naming this FilterOutNonPublicAccessor.  "filter" doesn't really mean anything on its own.